lib/shopify_api/rest/request.ex

```diff
@@ -162,4 +162,57 @@ defmodule ShopifyAPI.REST.Request do
          {:ok, body} <- map_fetched,
          do: body
   end
+
+  @doc """
+  Take an existing URI and add additional params, appending and replacing as necessary
+  ## Examples
+      iex> add_params_to_url("http://example.com/wat", [])
+      "http://example.com/wat"
+      iex> add_params_to_url("http://example.com/wat", [q: 1])
+      "http://example.com/wat?q=1"
+      iex> add_params_to_url("http://example.com/wat", [q: 1, t: 2])
+      "http://example.com/wat?q=1&t=2"
+      iex> add_params_to_url("http://example.com/wat", %{q: 1, t: 2})
+      "http://example.com/wat?q=1&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1&t=2", [])
+      "http://example.com/wat?q=1&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1", [t: 2])
+      "http://example.com/wat?q=1&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1", [q: 3, t: 2])
+      "http://example.com/wat?q=3&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1&s=4", [q: 3, t: 2])
+      "http://example.com/wat?q=3&s=4&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1&s=4", %{q: 3, t: 2})
+      "http://example.com/wat?q=3&s=4&t=2"
+  """
+  @spec add_params_to_url(binary, list) :: binary
+  def add_params_to_url(url, params) do
+    url
+    |> URI.parse()
+    |> merge_uri_params(params)
+    |> String.Chars.to_string()
```

Comment:
Credo is complaining about this because `Nested modules could be aliased at the top of the invoking module`. I disagree and think it starts to make sense to alias modules at the top when they occur in 2 places at the very least. How do other folks feel? Happy to `alias String.Chars` if credo is our one source of truth.

Reply:
We have been using Credo as our source of truth so we don't have to bike shed these kinds of things.

Reply:
Shouldn't even need the alias, just to_string() should work. Credo is also wrong sometimes.

Reply:
While we're here, though -- this function probably doesn't need to be part of the public API.